docs/data/material/components/grid2/grid2.md

```diff
@@ -130,21 +130,34 @@ The demo below shows how this works:
 
 ## Nested grid
 
-The grid container that renders inside another grid container is a nested grid that inherits its [`columns`](#columns) and [`spacing`](#spacing) from the top level.
+The grid container that renders as a **direct child** inside another grid container is a nested grid that inherits its [`columns`](#columns) and [`spacing`](#spacing) from the top level.
 It will also inherit the props of the top-level grid if it receives those props.
 
-### Inheriting columns
+:::success
 
-A nested grid container will inherits the columns from its parent unless the `columns` prop is specified to the instance.
+Note that a nested grid container should be a direct child of another grid container. If there are non-grid elements in between, the grid container will start as the new root container.
```

Comment:
This makes much more sense 👍 Note, we need to add a BREAKING CHANGE section in the PR description (and in the chalengelog entry).